.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,251 @@
+
+      - name: Get gradle wrapper and build dependencies
+        if: steps.cache-gradle.outputs.cache-hit != 'true'
+        run: |
+          ./configure.sh
+          ./gradlew --no-daemon dependencies
+
+      - name: Build
+        run: |
+          ./gradlew --no-daemon --stacktrace build -x test
+
+      - name: Archive build artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-files
+          path: |
+            rskj-core/build
+
+      - name: List the built jars
+        run: |
+          find rskj-core/build/libs/ -name "*.jar"
+
+  sonarqube:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - uses: actions/cache@v4
+        name: Restore Gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties', 'rskj-core/src/main/resources/version.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-files
+          path: |
+            rskj-core/build
+
+      - name: List the built artefacts
+        run: |
+          tree rskj-core/build -L 2  
+
+      - name: Run SonarQube analysis
+        run: |
+          extra_flags=""
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            git branch -f master origin/master
+            pr_number==${{ github.event.pull_request.number }}
+            extra_flags="-Dsonar.pullrequest.base=master
+              -Dsonar.pullrequest.branch=${{ github.head_ref }}
+              -Dsonar.pullrequest.key=$pr_number"
+          else
+            extra_flags="-Dsonar.branch.name=master"
+          fi
+          ./gradlew sonarqube --no-daemon -x build -x test \
+              $extra_flags \
+              -Dsonar.organization=rsksmart \
+              -Dsonar.host.url="https://sonarcloud.io" \
+              -Dsonar.token="${{ secrets.SONAR_TOKEN }}"
+
+  tests:
+    needs: build
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y curl
+
+      - uses: actions/cache@v4
+        name: Restore gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Unit tests
```

Comment:
I don't know how much value is added by changing this, but I would split this step into two different ones, with one running the unit tests and the other running the integration tests only.

With the second dependent on the first. Besides that, at the end is the same (and lasts for the same time) visually is better than having an action running for 30 minutes.

Reply:
Fair enough. Although I'm leaning more towards having them run in parallel to get a faster result for either.

Reply:
done